kb/2023-08-23/using_rook_ceph_storage.md

```diff
@@ -0,0 +1,86 @@
+slug: using_rook_ceph_storage
+authors:
+  - name: Hang Yu
+    title: Staff Software Engineer
+    url: https://github.com/futuretea
+    image_url: https://github.com/futuretea.png
+tags: [harvester, rook, ceph, csi]
+hide_table_of_contents: false
+---
+
+Harvester now supports installing a [Container Storage Interface (CSI)](https://kubernetes-csi.github.io/docs/introduction.html) in your Harvester cluster to support and use external storage as the VM's non-system data partition. Leverage different drivers for specific purposes, such as performance optimization or integration with existing internal storage providers.
+
+The provisioner of the VM image still uses Longhorn. Before v1.2.0, Harvester only supports using Longhorn to store virtual machine data and does not support using external storage to store VM data.
+
+[Rook](https://rook.io) is an open-source cloud-native storage orchestrator, providing the platform, framework, and support for Ceph storage to natively integrate with cloud-native environments.
+
+[Ceph](https://ceph.io) is a distributed storage system that provides file, block, and object storage and is deployed in large-scale production clusters.
+
+Rook automates deployment and management of Ceph to provide self-managing, self-scaling, and self-healing storage services. The Rook operator does this by building on Kubernetes resources to deploy, configure, provision, scale, upgrade, and monitor Ceph.
+
+This article explains how to install, configure, and use Rook Ceph as a data disk for a VM in Harvester.
+
+## Install Harvester Cluster
+
+Harvester's OS has an immutable design, which means most files in the  OS revert to their pre-configured state after a reboot.
+
+So we need to add some persistent paths used by Rook Ceph to `os.persistentStatePaths` in the [Harvester configuration](https://docs.harvesterhci.io/dev/install/harvester-configuration#ospersistent_state_paths).
+
+```yaml
+os:
+  persistentStatePaths:
+    - /var/lib/rook
+    - /var/lib/ceph
+  modules:
+    - rbd
+    - nbd
+```
+
+After the cluster is installed, refer to [How can I access the kubeconfig file of the Harvester cluster?](https://docs.harvesterhci.io/v1.1/faq#how-can-i-access-the-kubeconfig-file-of-the-harvester-cluster) to get the kubeconfig of the Harvester cluster.
+
+## Install Rook Ceph to Harvester
+
+Next, install Rook Ceph to the Harvester cluster by referring to the [Rook Ceph Quickstart](https://rook.io/docs/rook/v1.12/Getting-Started/quickstart/).
+
+1. Install Rook operator
+```bash
+curl -fsSLo rook.tar.gz https://github.com/rook/rook/archive/refs/tags/v1.12.2.tar.gz \
+  && tar -zxf rook.tar.gz && cd rook-1.12.2/deploy/examples
+# apply configurations ref: https://rook.github.io/docs/rook/v1.12/Getting-Started/example-configurations/
+kubectl apply -f crds.yaml -f common.yaml -f operator.yaml
+kubectl -n rook-ceph wait --for=condition=Available deploy rook-ceph-operator --timeout=10m
+```
```

Comment:
```suggestion
1. Install the Rook operator.

```bash
curl -fsSLo rook.tar.gz https://github.com/rook/rook/archive/refs/tags/v1.12.2.tar.gz \
  && tar -zxf rook.tar.gz && cd rook-1.12.2/deploy/examples
# apply configurations ref: https://rook.github.io/docs/rook/v1.12/Getting-Started/example-configurations/
kubectl apply -f crds.yaml -f common.yaml -f operator.yaml
kubectl -n rook-ceph wait --for=condition=Available deploy rook-ceph-operator --timeout=10m
```
```